apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/LogUploader.cs

```diff
@@ -109,60 +112,78 @@ private LogEntry MakeLostEntry(DateTimeRange lostRange)
                         logEntriesLost = _logEntriesLost;
                         _logEntriesLost = null;
                     }
-                    entries = await _logQ.PeekAsync(_maxUploadBatchSize - (logEntriesLost != null ? 1 : 0));
+                    entries = await _logQ.PeekAsync(_maxUploadBatchSize - (logEntriesLost != null ? 1 : 0), cancellationToken);
                     if (logEntriesLost != null)
                     {
                         var lostEntryExtra = new LogEntryExtra(-1, MakeLostEntry(logEntriesLost));
-                        entries = (new[] { lostEntryExtra }).Concat(entries ?? Enumerable.Empty<LogEntryExtra>());
+                        entries = (new[] { lostEntryExtra }).Concat(entries ?? Enumerable.Empty<LogEntryExtra>()).ToList();
                     }
                     if (entries != null && entries.Count() > 0)
                     {
                         // There are log entries that need uploading, so do that now.
                         break;
                     }
-                    await _uploadReadyEvent.WaitAsync();
+                    await _uploadReadyEvent.WaitAsync(cancellationToken);
                 }
                 // Upload entries to the Cloud Logging server
                 try
                 {
-                    await _client.WriteLogEntriesAsync(null, null, s_emptyLabels, entries.Select(x => x.Entry));
-                    await _logQ.RemoveUntilAsync(entries.Last().Id);
-                    _emptyEvent.Set();
+                    // TODO: We should be able to specify an empty log name here; all our log entries
```

Comment:
This TODO was removed deliberately - no need to reintroduce it :)
(That's what the null first argument is for.)